src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -2563,6 +2650,38 @@ private static ulong ExtractFractionAndBiasedExponent(double value, out int expo
             return fraction;
         }
 
+        private static ushort ExtractFractionAndBiasedExponent(Half value, out int exponent)
+        {
+            ushort bits = (ushort)BitConverter.HalfToInt16Bits(value);
+            ushort fraction = (ushort)(bits & 0x3FF);
+            exponent = ((int)(bits >> 10) & 0xFF);
```

Comment:
This needs to be `0x3F` to ensure the sign bit is masked out.
```suggestion
            exponent = ((int)(bits >> 10) & 0x3F);
```

Reply:
Good catch. Fixed